File: Fila.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Fila{

private:
    vector<int> fila;
    int tamanho;
    int inicio;
    int fim;
    
public:
    Fila(int tamanho){
        this->fila.resize(tamanho);
        this->tamanho = tamanho;
        this->inicio = 0;
        this->fim = 0;
    }

    void inserir(int valor) {
        if(fim - inicio < tamanho){ //nao ta vazia
            if(fim < tamanho){ //nao ta cheia
                fila[fim++] = valor;
            }
        }
    }

    void retirar() {
        if(inicio != fim){
            cout << fila[inicio++] << endl;
            if(inicio == fim){
                inicio = 0;
                fim = 0;
            }
        }
    }

};

int main(){

    int tamanho;
    cin >> tamanho;
    
    Fila fila(tamanho);

    string op;

    while(cin >> op){

        if(op == "E"){
            int valor;
            cin >> valor;
            fila.inserir(valor);
        }
        else if (op == "D"){
            fila.retirar();
        }  
    }

    return 0;
}

File: Fila.java

import java.util.*;

public class Fila {

    private int[] fila;
    private int tamanho;
    private int inicio; //retira
    private int fim; //insere

    //fila tem tamanho, inicio e fim
    public Fila(int tamanho){
        this.fila = new int[tamanho];  
        this.tamanho = tamanho;
        this.inicio = 0;
        this.fim = 0;            
    }

    public void inserir(int valor){
        if(fim - inicio < tamanho){
            if(fim < tamanho){
                fila[fim++] = valor;
            }
        }
    }

    public void retirar(){
        if(inicio != fim){
            System.out.println(fila[inicio++]);
            if(inicio == fim){
                inicio = 0;
                fim = 0;
            }
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();

        Fila fila = new Fila(N);

        while(scanner.hasNext()){
            String op = scanner.next();
            if("E".equals(op)){
                int valor = scanner.nextInt();
                fila.inserir(valor);
            }
            else if("D".equals(op)){
                fila.retirar();
                }
        }
    }
    
}
